src/imu/imu.cpp

```diff
@@ -0,0 +1,115 @@
+/****************************************************************
+ * Example1_Basics.ino
+ * ICM 20948 Arduino Library Demo
+ * Use the default configuration to stream 9-axis IMU data
+ * Owen Lyke @ SparkFun Electronics
+ * Original Creation Date: April 17 2019
+ *
+ * Please see License.md for the license information.
+ *
+ * Distributed as-is; no warranty is given.
+ * 
+ * Modified by Emily Thorpe - Auguest 2023
+ ***************************************************************/
+#include "ICM_20948.h" 
+#include "cli/conio.hpp"
+#include "cli/flog.hpp"
+
+#define SERIAL_PORT Serial
+
+#define WIRE_PORT Wire
+#define AD0_VAL 1
+
+ICM_20948_I2C myICM; 
+
+
+float getAccMG( int16_t raw, uint8_t fss );
+float getGyrDPS( int16_t raw, uint8_t fss );
+float getMagUT( int16_t raw );
+float getTmpC( int16_t raw );
+
+void setupICM(void)
+{
+    WIRE_PORT.begin();
+    WIRE_PORT.setClock(400000);
+
+    myICM.begin(WIRE_PORT, AD0_VAL);
+
+    SF_OSAL_printf("Initialization of the sensor returned: ");
+    SF_OSAL_printf(myICM.statusString());
+    if (myICM.status != ICM_20948_Stat_Ok)
+    {
+        SF_OSAL_printf("ICM fail");
+        FLOG_AddError(FLOG_ICM_FAIL, 0);
+    }
+}
+
+
+bool getAccelerometer(float *acc_x, float *acc_y, float *acc_z)
+{
+    ICM_20948_AGMT_t agmt = myICM.getAGMT();
+    
+    *acc_x = getAccMG(agmt.acc.axes.x, agmt.fss.a);
+    *acc_y = getAccMG(agmt.acc.axes.y, agmt.fss.a);
+    *acc_z = getAccMG(agmt.acc.axes.z, agmt.fss.a);
+
+    return true;
+}
+
+bool getGyroscope(float *gyr_x, float *gyr_y, float *gyr_z)
+{
+    ICM_20948_AGMT_t agmt = myICM.getAGMT();
+    
+    *gyr_x = getGyrDPS(agmt.gyr.axes.x, agmt.fss.g);
+    *gyr_y = getGyrDPS(agmt.gyr.axes.y, agmt.fss.g);
+    *gyr_z = getGyrDPS(agmt.gyr.axes.z, agmt.fss.g);
+
+    return true;
+}
+
+bool getMagnetometer(float *mag_x, float *mag_y, float *mag_z)
+{
+    ICM_20948_AGMT_t agmt = myICM.getAGMT();    
+
+    *mag_x = getMagUT(agmt.mag.axes.x);
+    *mag_y = getMagUT(agmt.mag.axes.y);
+    *mag_z = getMagUT(agmt.mag.axes.z);
+
+    return true;
+}
+
+bool getTemperature(float *temperature)
+{
+    ICM_20948_AGMT_t agmt = myICM.getAGMT();
+
+    *temperature = getTmpC(agmt.tmp.val);
+    return true;
+}
```

Comment:
What units is this in?

Reply:
I've read through  the datasheet, and I cant seem to find that.

Reply:
is this in deg C (See `getTmpC`)